### DISCUSSION TITLE
Airflow task getting aborted(Negsignal.SIGABRT) when snowflake connection attempted

### DISCUSSION THREAD

**Discussion by saurabh782** [apache/airflow#39902] [Q&A]
Hi guys,
I am running airflow on local and trying to execute a dag. The dag connects to snowflake. Whenever the connection is attempted the task terminates with the following error.

`[2024-05-29, 04:57:52 IST] {local_task_job.py:154} INFO - Task exited with return code Negsignal.SIGABRT`

Attaching the entire log:

> `[2024-05-29, 04:57:51 IST] {snowflake_util.py:119} INFO - Cannot access SSM parameter using *** from Config to authenticate Snowflake
> [2024-05-29, 04:57:51 IST] {connection.py:275} INFO - Snowflake Connector for Python Version: 2.7.0, Python Version: 3.7.9, Platform: Darwin-23.2.0-x86_64-i386-64bit
> [2024-05-29, 04:57:51 IST] {connection.py:891} INFO - This connection is in OCSP Fail Open Mode. TLS Certificates would be checked for validity and revocation status. Any other Certificate Revocation related exceptions or OCSP Responder failures would be disregarded in favor of connectivity.
> [2024-05-29, 04:57:51 IST] {connection.py:908} INFO - Setting use_openssl_only mode to False
> [2024-05-29, 04:57:52 IST] {local_task_job.py:154} INFO - Task exited with return code Negsignal.SIGABRT`

snowflake-connector-python==2.7.0
apache-airflow==2.2.3
python==3.7.9

Now I referred to the stack overflow question which has almost same issue: https://stackoverflow.com/questions/66012040/task-exited-with-return-code-negsignal-sigabrt-airflow-task-fails-with-snowflak

But I tried the suggestion of `export OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES` its not working for me. 

Few more details when I run it as a standalone python script it runs completely fine. But when I run it as an airflow dag the task terminates.

Can someone pls help?

**Comment by Taragolis** [comment]
Difficult to say, ``SIGABRT`` it something which difficult to trace without additional information

  - Set environment variable [`PYTHONFAULTHANDLER`](https://docs.python.org/3/library/faulthandler.html) to `true`.
  - (Re)Start airflow services.
  - Restart failed airflow task.
  - Check 'scheduler' and 'worker' services logs for additional traceback
    which might contain information about module/library where actual error happen.
